source/material_model/rheology/diffusion_creep.cc

```diff
@@ -91,11 +105,34 @@ namespace aspect
         // A: prefactor,
         // d: grain size, m: grain size exponent, E: activation energy, P: pressure,
         // V; activation volume, R: gas constant, T: temperature.
-        double viscosity_diffusion = 0.5 / p.prefactor *
-                                     std::exp((p.activation_energy +
-                                               pressure*p.activation_volume)/
-                                              (constants::gas_constant*temperature)) *
-                                     std::pow(grain_size, p.grain_size_exponent);
+
+	double viscosity_diffusion = 0;
+
+	if (use_water_fugacity_diffusion)
+	  {
+	     AssertThrow(this->introspection().compositional_name_exists("water"),
```

Comment:
I'm not sure "water" is a good name for the compositional field. Water is a free phase, not a chemical component in a solid. Perhaps "H2O concentration" would be more appropriate?